include/soci/type-conversion.h

```diff
@@ -28,7 +28,12 @@ namespace details
 template <typename T>
 struct base_value_holder
 {
+    base_value_holder()
```

Comment:
In principle, this code is not type-dependent, so it might be extracted into some `base_value_holder_type_indep` subclass from which `base_value_holder` would inherit to reduce "template bloat". I admit I'm not really sure if it's worth doing it in this case, but it might be.